File: C2109L/ADF1/15-09-2022/javaPE01/src/main/java/com/aptech/inheritance/Rectangle.java
package com.aptech.inheritance;

public class Rectangle extends Shape{
    //it can getName, getColor
    private int width, height;

    public Rectangle(String name, String color,
                     int width, int height) {
        super(name, color);//extends from super class
        this.width = width;
        this.height = height;
    }



    public int getWidth() {
        return width;
    }

    public void setWidth(int width) {
        this.width = width;
    }

    public int getHeight() {
        return height;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    @Override
    public int getArea() {
        //You MUST implement abstract method from super class
        return width * height;
    }

    @Override
    public String toString() {
        return super.toString() + "\n"+
                "width: "+width +"\n"+
                "height: "+height +"\n" +
                "area: "+this.getArea();
    }
}

File: C1908iLeThanhNghi/09-10-2020/09-10-2020/09-10-2020/Rectangle.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace _09_10_2020
{
    class Rectangle:Shape
    {
        public double Width { get; set; }
        public double Height { get; set; }

        public override void ShowArea()
        {
            Console.WriteLine($"Area = {Width * Height}");
        }
    }
}
